sdc/datatypes/common_functions.py

```diff
@@ -438,6 +449,34 @@ def hpat_join_series_indexes_impl(left, right):
 
             return joined, lidx, ridx
 
-        return hpat_join_series_indexes_impl
+        return sdc_join_series_indexes_impl
 
     return None
+
+
+def sdc_check_indexes_equal(left, right):
+    pass
+
+
+@overload(sdc_check_indexes_equal)
+def sdc_check_indexes_equal_overload(A, B):
+    """Function for checking arrays A and B of the same type are equal"""
+
+    if isinstance(A, types.Array):
+        def sdc_check_indexes_equal_numeric_impl(A, B):
+            return numpy.array_equal(A, B)
+        return sdc_check_indexes_equal_numeric_impl
+
+    elif A == string_array_type:
+        def sdc_check_indexes_equal_string_impl(A, B):
+            # TODO: replace with StringArrays comparison
+            is_index_equal = (len(A) == len(B)
+                              and num_total_chars(A) == num_total_chars(B))
+            for i in numpy.arange(len(A)):
```

Comment:
Perhaps it makes sense to use `numba.prange` instead of `numpy.arange`.

Reply:
@densmirn I don't think we can use prange here, as multiple threads will write into the same is_index_equal variable.